bindings/c/include/opendal.h

```diff
@@ -17,25 +17,33 @@
  * under the License.
  */
 
+#ifndef __OPENDAL_H__
+#define __OPENDAL_H__
 
-#ifndef _OPENDAL_H
-#define _OPENDAL_H
+/* Do not modify. This code was autogenerated by bindgen. */
 
-#include <stdint.h>
-#include <stddef.h>
+#include <stdarg.h>
 #include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
 
-#ifdef __cplusplus
-extern "C" {
-#endif // __cplusplus
+typedef struct BlockingOperator BlockingOperator;
 
-/*
- Hello, OpenDAL!
+typedef struct DynArray {
+  uint8_t *array;
+  size_t length;
+} DynArray;
+
+/**
+ * Hello, OpenDAL!
  */
 void hello_opendal(void);
 
-#ifdef __cplusplus
-} // extern "C"
-#endif // __cplusplus
+struct BlockingOperator *blocking_op_create(void);
+
+struct DynArray blocking_read(const struct BlockingOperator *op, const char *path);
+
+void blocking_write(const struct BlockingOperator *op, const char *path, const struct DynArray *bs);
```

Comment:
Is it common for C to use `DynArray` to write data?

https://github.com/awslabs/aws-c-s3/blob/main/include/aws/s3/s3_client.h#L103-L105